src/components/pathway/PathwayModal.jsx

```diff
@@ -0,0 +1,200 @@
+import React, { useContext } from 'react';
+import PropTypes from 'prop-types';
+import { Link } from 'react-router-dom';
+import Skeleton from 'react-loading-skeleton';
+import { AppContext } from '@edx/frontend-platform/react';
+import {
+  Icon, MarketingModal, ModalDialog, Container, Row, Col, Collapsible, Button, Image,
+} from '@edx/paragon';
+import { Assignment, BookOpen, VerifiedBadge } from '@edx/paragon/icons';
+import { useLearnerPathwayData } from './data/hooks';
+
+const coursesAndProgramsText = pathway => {
+  let courseCount = 0;
+  let programCount = 0;
+  let textCourse = '';
+  let textProgram = '';
+  let message;
+
+  for (let i = 0; i < pathway.steps.length; i++) {
+    courseCount += pathway.steps[i].courses?.length;
+    programCount += pathway.steps[i].programs?.length;
+  }
+
+  textCourse = courseCount === 1 ? 'course' : 'courses';
+  textProgram = programCount === 1 ? 'program' : 'programs';
+
+  if (courseCount && programCount) {
```

Comment:
nit: can this be
``
message = `${courseCount && `${courseCount} ${textCourse}`} ${programCount && `and ${programCount} ${textProgram}`}`
``

Reply:
isn't this very hard to understand?